daffodil-core/src/test/scala/org/apache/daffodil/dsom/TestPolymorphicUpwardRelativeExpressions.scala

```diff
@@ -144,8 +144,8 @@ class TestPolymorphicUpwardRelativeExpressions extends Logging {
     }
     val msg = e.getMessage()
     val hasSDE = msg.contains("Schema Definition Error")
-    val hasPolyInt = msg.contains("../../poly with xs:int")
-    val hasPolyDate = msg.contains("../../poly with xs:date")
+    val hasPolyInt = msg.contains("../../poly eq 5 with xs:int")
+    val hasPolyDate = msg.contains("../../poly eq 5 with xs:date")
```

Comment:
What is the full error here? Without context of the rest of the error message it's hard to tell if this change is correct.

Reply:
The addition of 'eq 5' is what the wholeText returns.

Reply:
Right, but the full message is probably someting like "Expression failed because ..... some more text ... ../../poly eq 5 with xs:date ... maybe more text". The on snipp of ``../../poly eq 5 with xs:date`` feels like it doesn't make sense, but maybe it's because I don't have the full context of the full message. I was asking what the full message is to make sure this error message still makes sense with this change, since it isn't clear.

Reply:
Oh, here we go:
Schema Definition Error: Subset Expression ../../poly has different types at different points of usage.
The inconsistent type usages are listed here:
element poly in expression ../../poly eq 5 with xs:int type at Location line 29 column 10 in file:/C:/Users/RSTEIN~1/AppData/Local/Temp/anon689374424817542066.dfdl.xsd
element poly in expression ../../poly eq 5 with xs:date type at Location line 48 column 10 in

Reply:
Gotcha, that seems like a reasonble (and more helpful with this change) diagnostic to me. Looks good.

Reply:
Cool